extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/ReadMongoP.java

```diff
@@ -349,7 +349,7 @@ private boolean hasSorts(List<Bson> aggregateList) {
 
         private Traverser<Document> delegateForCollection(MongoCollection<Document> collection,
                                                           List<Bson> aggregateList) {
-            return traverseIterable(collection.aggregate(aggregateList));
+            return traverseIterable(collection.aggregate(aggregateList).batchSize(BATCH_SIZE));
```

Comment:
this change is unrelated but probably desired

Reply:
Yes, discovered in the meantime that here I was missing this method. It didn't break functionality, the performance was sub-par.